### PR TITLE
Implement support for AsyncIterator

### LINKED ISSUE #12: Add API for handling multiple events by returning an async iterator
As an addition to #3. This is a more low-level primitive.

http://2ality.com/2016/10/asynchronous-iteration.html

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
Proposed to close #12.  The tests are pretty simple and the documentation has to be written.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -65,3 +67,125 @@ module.exports = (emitter, event, options) => {
...
+
+	let addListener = emitter.on || emitter.addListener || emitter.addEventListener;
+	let removeListener = emitter.off || emitter.removeListener || emitter.removeEventListener;
+
+	if (!addListener || !removeListener) {
+		throw new TypeError('Emitter is not compatible');
+	}
+
+	addListener = addListener.bind(emitter);
+	removeListener = removeListener.bind(emitter);
```
Comment (sindresorhus) [inline index.js]: Line 81 to 89 could be shared with the main export implementation by abstracting it into a method that accepts the emitter and returns the `addListener` and `removeLister` methods.

**Comment by sindresorhus** [comment]
This looks very good. Exactly what I had in mind :)

This will need some docs. I would be happy to do that part unless you want to?

File: index.js [hunk 2]
```diff
@@ -65,3 +67,125 @@ module.exports = (emitter, event, options) => {
...
 	return ret;
 };
+module.exports.iterator = (emitter, event, options) => {
+	if (typeof options === 'function') {
+		options = {filter: options};
+	}
+
+	options = Object.assign({
+		rejectionEvents: ['error'],
+		resolutionEvents: [],
```
Comment (sindresorhus) [inline index.js]: Should we have any default names here, like `['end', 'finish']`?

**Comment by sindresorhus** [comment]
[I added some docs](https://github.com/sindresorhus/p-event/pull/14/commits/694199c3e43c4a9fea4569fc6c4bf2a06c3441ce). Let me know what you think. Feel free to improve them.

File: index.js (hunk 2 above)
Comment (wtgtybhertgeghgtwtg) [reply index.js]: I wouldn't.  I think the default for error is a special case, since `node` will throw a fit if you don't handle 'error' events.

**Comment by sindresorhus** [comment]
Nice work, @wtgtybhertgeghgtwtg 🙌

**Pull request merged**